spec/models/course_spec.rb

```diff
@@ -129,54 +135,73 @@
 
     describe '#has_vacancies?' do
       context 'for a single site status that has vacancies' do
-        let(:subject) {
-          create(:course, with_site_statuses: [%i[findable applications_being_accepted_now with_any_vacancy]])
-        }
+        before do
+          findable_site_status
+          applications_being_accepted_now_site_status
+          site_status_with_any_vacancy
+        end
+
+        subject { course }
 
         its(:has_vacancies?) { should be true }
       end
 
       context 'for a site status with vacancies and others without' do
-        let(:subject) {
-          create(:course, with_site_statuses: [
-                   %i[findable applications_being_accepted_now with_any_vacancy],
-                   %i[findable with_no_vacancies],
-                   %i[findable with_no_vacancies],
-                 ])
+        let(:site_status_with_vacancies) {
+          create(:site_status,
+                 :findable,
+                 :with_any_vacancy,
+                 :applications_being_accepted_now)
+        }
+        let(:findable_site_status_with_no_vacancies) {
+          create(:site_status,
+                 :findable,
+                 :no_vacancies,
+                 course: site_status_with_vacancies.course)
         }
 
+        before do
+          site_status_with_vacancies
+          findable_site_status_with_no_vacancies
+          findable_site_status_with_no_vacancies
```

Comment:
This doesn't actually do the same as the original ... it looks like it creates two `site_statuses`, but it doesn't since `let` gives us memoized values. BUT ... I can't see why we needed two in the first place, so how about we just get rid of one and call it a day?

Reply:
Good shout. Have removed the duplicate findable site status.